### PR TITLE
Major Project cleanup: Xcode 10, Swift 4.2, PromiseKit 6, etc.

### PR DESCRIPTION
**Pull request by freak4pc**
I saw Hacktoberfest starting and was interested in some of the OSS work you do at Microsoft. Was looking for some iOS/Swift related task to do and noticed there are several issues with this project:

- It’s highly not updated (Still using Swift 3 and old dependencies).
- Many pieces of the code needed some love.
- Formatting and style guides are not very much according to any standard style guide as of today. 

This PR couldn’t possibly resolve all issues in this project but I did do sort of a “wipe all” with a pretty big refactor around most of the files to try and modernize some of the parts that didn’t require re-writing the entire thing.

Basically this PR:

- Cleans up some of the basic files and remove unneeded / unused code.
- Updates formatting and styling according to generally accepted community Swift guidelines.
- Go over all of the files and try to cleanup the implementation of anything that wouldn’t require a rewrite / too long :) 
- Bump dependencies. 
- Update for Swift 4.2 & Xcode 10.

There are still some issues in the project that could be resolved later such as - a lot of the UI code is heavily coupled with the network, etc. but I imagine that you might not be bothered by that for a demo project.

Quite a bit of changes, hope this is not too overwhelming or too much of a change. Will be happy to answer any feedback or questions.

Cheers!
Shai.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -41,7 +41,6 @@ DerivedData
 # CocoaPods
 #
 Pods/
-*Podfile.lock
```
Comment (freak4pc) [inline .gitignore:44]: https://guides.cocoapods.org/using/using-cocoapods#benefits-of-ignoring-the-pods-directory

> Whether or not you check in the Pods directory, the Podfile and **Podfile.lock** should always be kept under version control.

**Comment by freak4pc** [comment]
Anyone maintaining this repo? 
Perhaps @jthake-msft ?

**Comment by jthake** [comment]
@freak4pc thanks for submitting this! I will review this on my Mac in the evening this week!

@jasonjoh do you think you could do a review too? I know this is at odds with our other work of consolidating samples. But I wonder whether there is some things we should take on from this into the new work we're shipping?

File: .gitignore (hunk 1 above)
Comment (jasonjoh) [reply .gitignore:44]: Agreed :)

**Review by jasonjoh — APPROVED** [review]
Looks good, thanks for doing this.

**Comment by freak4pc** [comment]
Sure thing. Appreciate your review 👍

**Comment by freak4pc** [comment]
@jthake-msft I think this could be merged given it was reviewed?

**Comment by freak4pc** [comment]
I'm 4/5 ;-) Just saying' ... :P 

image

**Comment by jasonjoh** [comment]
Well we can't have that! :)

**Pull request merged**